lambdas/evaluate-gpg45-scores/src/test/java/uk/gov/di/ipv/core/evaluategpg45scores/EvaluateGpg45ScoresHandlerTest.java

```diff
@@ -486,6 +495,77 @@ void shouldReturnJourneyMetIfCheckRequiresAdditionalEvidenceResponseFalse() thro
         verify(userIdentityService, times(1)).checkRequiresAdditionalEvidence(any());
     }
 
+    @Test
+    void shouldReturnJourneyMetForLowConfidence() throws Exception {
+        when(ipvSessionService.getIpvSession(TEST_SESSION_ID)).thenReturn(ipvSessionItem);
+        when(gpg45ProfileEvaluator.getFirstMatchingProfile(any(), eq(P1_PROFILES)))
+                .thenReturn(Optional.of(L1A));
+        clientOAuthSessionItem.setVtr(List.of("P1"));
+        when(clientOAuthSessionDetailsService.getClientOAuthSession(any()))
+                .thenReturn(clientOAuthSessionItem);
+        when(userIdentityService.areVcsCorrelated(any())).thenReturn(true);
+        when(userIdentityService.checkRequiresAdditionalEvidence(any())).thenReturn(false);
+
+        JourneyResponse response =
+                toResponseClass(
+                        evaluateGpg45ScoresHandler.handleRequest(request, context),
+                        JourneyResponse.class);
+
+        assertEquals(JOURNEY_MET.getJourney(), response.getJourney());
+
+        assertEquals(Vot.P1, ipvSessionItem.getVot());
+
+        ArgumentCaptor<AuditEvent> auditEventCaptor = ArgumentCaptor.forClass(AuditEvent.class);
+        verify(auditService).sendAuditEvent(auditEventCaptor.capture());
+        AuditEvent auditEvent = auditEventCaptor.getValue();
+        AuditExtensionGpg45ProfileMatched extension =
+                (AuditExtensionGpg45ProfileMatched) auditEvent.getExtensions();
+        assertEquals(L1A, extension.getGpg45Profile());
+    }
+
+    @Test
+    void shouldReturnJourneyMetForMeetingMediumAndLowConfidences() throws Exception {
+        when(ipvSessionService.getIpvSession(TEST_SESSION_ID)).thenReturn(ipvSessionItem);
+        clientOAuthSessionItem.setVtr(List.of("P1", "P2"));
+        when(clientOAuthSessionDetailsService.getClientOAuthSession(any()))
+                .thenReturn(clientOAuthSessionItem);
+        when(gpg45ProfileEvaluator.getFirstMatchingProfile(any(), eq(P1_AND_P2_PROFILES)))
+                .thenReturn(Optional.of(M1A));
+        when(userIdentityService.areVcsCorrelated(any())).thenReturn(true);
+        when(userIdentityService.checkRequiresAdditionalEvidence(any())).thenReturn(false);
+
+        JourneyResponse response =
+                toResponseClass(
+                        evaluateGpg45ScoresHandler.handleRequest(request, context),
+                        JourneyResponse.class);
+
+        assertEquals(JOURNEY_MET.getJourney(), response.getJourney());
+        verify(ipvSessionItem).setVot(Vot.P2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("vcsMatchingVtr")
+    void shouldReturnTrueForMatchingGpg45Profile(List<String> vtr, List<Gpg45Profile> profiles)
```

Comment:
I'm not sure this test name describes what this is testing?
It looks like it's testing that the correct set of profiles are matched against for a given vtr

Reply:
Also can we test this without making `hasMatchingGpg45Profile` public?

Reply:
> I'm not sure this test name describes what this is testing?

I'll rename to `shouldReturnTrueForHasMatchingProfileGivenProfilesMatchVtr`?

> Also can we test this without making hasMatchingGpg45Profile public?

I already have but it wasn't as clear as I'd hoped, since the `handleRequest` tests have so to them already, so I wanted to clarify with a more explicit test. I'll remove if you think it's best.

Reply:
> I'll rename to shouldReturnTrueForHasMatchingProfileGivenProfilesMatchVtr?
Calling it `shouldReturnTrue...` suggests that we should be checking that the method returns true, but the assert section is testing that the correct profiles were supplied to `getFirstMatchingProfile` - no `true` in sight :)

Reply:
Ah good point I've removed the check for the result. I'll rename the method source as well

`shouldUseCorrectProfilesFromVtrToCheckMatchingGpg45Profile`